File: c-interop/native-raw/src/commonMain/c/crypto.c
#include <openssl/crypto.h>

const void* ffi_OpenSSL_version_call __asm("ffi_OpenSSL_version");
const void* ffi_OpenSSL_version_call = (const void*)&OpenSSL_version;

const void* ffi_OPENSSL_version_major_call __asm("ffi_OPENSSL_version_major");
const void* ffi_OPENSSL_version_major_call = (const void*)&OPENSSL_version_major;

File: c-interop/native-raw/src/commonMain/kotlin/crypto.kt
@file:Suppress("FunctionName")

package dev.whyoleg.kip.c.native.raw

import kotlinx.cinterop.internal.*

// magic :)
// defined as `# define OPENSSL_VERSION_STRING 6`
const val OPENSSL_VERSION_STRING = 6

@CCall("ffi_OpenSSL_version")
external fun OpenSSL_version(type: Int): Pointer_Byte

@CCall("ffi_OPENSSL_version_major")
external fun OPENSSL_version_major(): Int
